Testing/Cxx/basic_SYCL_test.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file
+ * @brief Basic test to check that SYCL is running correctly on the selected device ("gpu", "cpu")
+ * @author Declan Garvey (IFIC, CSIC-UV)
+ */
+
+#include <CLI/CLI.hpp>
+#include <iostream>
+#include <string>
+#include <sycl/sycl.hpp>
+
+/**
+ * @brief Basic function to free memory that was allocated during the test
+ * @param q The sycl queue used
+ * @param testArrayOnDevice The unsigned int array used for testing
+ * @param submitFlag Error flag to signal unsuccessful submiting
+ * @param copyFlag Error flag to signal unsuccessful copying
+ * @param arithmeticFlag Error flag to signal unsuccessful arithmetics
+ */
+void cleanup(sycl::queue &q, unsigned *testArrayOnDevice, bool *submitFlag, bool *copyFlag,
+             bool *arithmeticFlag)
+{
+  if (testArrayOnDevice)
+    sycl::free(testArrayOnDevice, q);
+  if (submitFlag)
+    sycl::free(submitFlag, q);
+  if (copyFlag)
+    sycl::free(copyFlag, q);
+  if (arithmeticFlag)
+    sycl::free(arithmeticFlag, q);
+}
+
+/**
+ * @brief main
+ * @param argc Argument counts
+ * @param argv Variables
+ * @return EXIT_FAILURE if error, EXIT_SUCCESS if ok
+ */
+int main(int argc, char **argv)
+{
+  CLI::App app{"Simple ctest to check if SYCL is functioning correctly"};
+  std::string deviceName = "gpu";
+  app.add_option<std::string>("-d,-D,--DeviceName", deviceName,
+                              "The name of the device to run test on, Accepts: gpu or cpu");
+  CLI11_PARSE(app, argc, argv);
+
+  // Create SYCL queue
+  sycl::queue q;
+  if (deviceName == "cpu") {
+    q = sycl::queue{sycl::multi_cpu_selector_v};
```

Comment:
```suggestion
    q = sycl::queue{sycl::cpu_selector_v};
```
seems this was not part of the SYCL standard but an adaptivecpp extension